### DISCUSSION TITLE
Add grouping by properties/prefixes

### DISCUSSION THREAD

**Discussion by ghost** [tailwindlabs/tailwindcss#12155] [Ideas]
Currently, **tailwind** requires that for each property with a prefix, you substitute that prefix. Or for each property with a media query, you must supply the class of that media query. Same for CSS properties that accept multiple parameters:

`after:block after:h-16 after:w-16 after:animate-spin`

or

`transition-colors or transition-[color]` (without the ability to list properties to be animated)

or (for media queries)

`md:h-2 md:h-[20px] md:flex`

It would be cool to group such properties to eliminate repetition and, best of all, to make writing styles more understandable as it is currently implemented in CSS/SCSS where a selector block contains CSS properties. And the KISS principle. Example:

**So it can look accordingly:**

`after:[block h-16 w-16 animate-spin]`

_or_

`transition-[color background-color border-color]`

_or (for media queries)_

`md:[h-2 h-[20px] flex]`

You get the idea. I don't know if there is something similar now, but I haven't found it

**Comment by wongjn** [comment]
There have been suggestions about adding such a feature. The maintainers have addressed this request before, that you may wish to read up on if you have not already done so:
- [A Twitter thread from the creator of Tailwind, Adam Wathan](https://twitter.com/adamwathan/status/1461519820411789314)
- [A previous GitHub discussion thread with maintainer responses](https://github.com/tailwindlabs/tailwindcss/discussions/8337#discussioncomment-4032611)

**Reply by ghost** [reply]
Hey @wongjn, thanks for the links, didn't see them and didn't realize there was such a discussion about it :)
